app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java

```diff
@@ -46,33 +56,42 @@ void setInteractor(GridFirestoreInteractor interactor) {
 
     @Override
     public void registerLocation(Carrier carrier, Location location, Date time) {
-        refreshLastPositions(time,location);
-        gridInteractor.write(location, String.valueOf(time.getTime()), carrier, successListener, failureListener);
+        registerLocation(carrier, location, time, successListener, failureListener);
     }
+
     @Override
     public void registerLocation(Carrier carrier,
                                  Location location,
                                  Date time,
                                  OnSuccessListener successListener,
-                                 OnFailureListener failureListener){
-        refreshLastPositions(time,location);
-        gridInteractor.write(location, String.valueOf(time.getTime()), carrier, successListener, failureListener);
+                                 OnFailureListener failureListener) {
+        refreshLastPositions(time, location);
 
+        Map<String, Object> element = new HashMap<>();
+        element.put("geoPoint", new GeoPoint(location.getLatitude(), location.getLongitude()));
+        element.put("timeStamp", time.getTime());
+        element.put("infectionStatus", carrier.getInfectionStatus());
+        gridInteractor.writeDocumentWithID("LastPositions", AccountFragment.getAccount(getActivity()).getId(),
+                element, s -> {
+                    gridInteractor.write(location, String.valueOf(time.getTime()), carrier, successListener, failureListener);
```

Comment:
Oh and the gridInteractor internaly already calls a writeDocumentWithID and a writeDocument. You made it wrong, please only use gridInteractor.writeDocumentWithID and not write (which is not what you want to do) OR USE : gridInteractor.write ONLY
Maybe I got it wrong but I don't think so

Reply:
The code with .write was already there when I implemented the feature so I assumed it was correct.

This could be an improvement for later if you're right

Reply:
Ok my bad, actually we do store on 2 different locations, thus use twice a write.